yash-datalake/nested/elasticSearchKibana.json

```diff
@@ -0,0 +1,326 @@
+                            "protocol": "*",
+                            "access": "Allow",
+                            "direction": "Inbound",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "7474"
+                        }
+                    },
+                    {
+                        "name": "neo4j-outbound-access",
+                        "properties": {
+                            "priority": 120,
+                            "protocol": "*",
+                            "access": "Allow",
+                            "direction": "Outbound",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "7474"
+                        }
+                    },
+                    {
+                        "name": "neo4j-bolt-inbound-access",
+                        "properties": {
+                            "priority": 1040,
+                            "protocol": "*",
+                            "access": "Allow",
+                            "direction": "Inbound",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "7687"
+                        }
+                    },
+                    {
+                        "name": "neo4j-bolt-outbound-access",
+                        "properties": {
+                            "priority": 130,
+                            "protocol": "*",
+                            "access": "Allow",
+                            "direction": "Outbound",
+                            "sourceAddressPrefix": "*",
+                            "sourcePortRange": "*",
+                            "destinationAddressPrefix": "*",
+                            "destinationPortRange": "7687"
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines/extensions",
+            "name": "[concat(variables('vmName'),'/', variables('extensionName'))]",
+            "apiVersion": "2018-10-01",
+            "location": "[parameters('location')]",
+            "properties": {
+                "publisher": "Microsoft.Azure.Extensions",
+                "type": "CustomScript",
+                "typeHandlerVersion": "2.0",
+                "autoUpgradeMinorVersion": true,
+                "settings": {
+                    "fileUris": [
+                        "[uri(variables('_artifactsLocation'),concat(variables('scriptFolder'),'/',variables('scriptFilename')))]"

```

Comment:
will need to add the sasToken

Reply:
Currently our scripts for installing Elastic Search and Kibana are hosted on Github where we do not have any SaS token available. Even, If we try to add SaS token in yash-datalake/nested/elasticSearchKibana.json template with empty value, we use to get a validation exception during Azure CI/CD process where it says empty parameters are not allowed. Also, as per your earlier comments we should not be having any empty variables/parameters.
Please guide here, how should we pass SaS token here ??

Reply:
you need to follow this guidance precisely: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts

Reply:
Added SAS Token to the uri